context/context.go

```diff
@@ -350,62 +356,56 @@ func (k Context) Pool() *pgxpool.Pool {
 // KubeAuthFingerprint generates a unique SHA-256 hash to identify the Kubernetes API server
 // and client authentication details from the REST configuration.
 func (k *Context) KubeAuthFingerprint() string {
-	rs := k.KubernetesRestConfig()
-	if rs == nil {
+	kc, _ := k.Kubernetes()
+	if kc == nil {
 		return ""
 	}
-
-	return hash.Sha256Hex(fmt.Sprintf("%s/%s/%s/%s/%s/%s",
-		rs.Host,
```

Comment:
Might be missing APIserver path etc.. 

This hash doesn't uniquely identify the server, if connections with the same server, but a different bearer token should have the same hash and use the same cache.